Tests/CloudPak4DataAuthTests.cs

```diff
@@ -0,0 +1,209 @@
+/**
+* Copyright 2021 IBM Corp. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*/
+
+using System.Collections;
+using System.Collections.Generic;
+using NUnit.Framework;
+using IBM.Cloud.SDK.Connection;
+using IBM.Cloud.SDK.Authentication;
+using IBM.Cloud.SDK.Authentication.Cp4d;
+
+namespace IBM.Cloud.SDK.Tests
+{
+    public class CloudPak4DataAuthTests
+    {
+        [Test]
+        public void TestConstructionRequried()
+        {
+            var url = "http://www.service-endpoint.com";
+            var username = "username";
+            var password = "password";
+            CloudPakForDataAuthenticator authenticator = new CloudPakForDataAuthenticator(
+                url: url,
+                username: username,
+                password: password);
+
+            Assert.IsNotNull(authenticator);
+            Assert.IsTrue(authenticator.AuthenticationType == Authenticator.AuthTypeCp4d);
+            Assert.IsTrue(authenticator.Url == url);
```

Comment:
Wouldn't this test result in false since the `authenticator.Url` would be http://www.service-endpoint.com/v1/authorize while `url` is http://www.service-endpoint.com

Reply:
authenticator.Url is `http://www.service-endpoint.com`. That doesn't change in the code, the connector.Url does though, but it's different